src/main/java/hudson/plugins/checkstyle/CheckStyleResult.java

```diff
@@ -71,6 +128,15 @@ protected CheckStyleResult(final AbstractBuild<?, ?> build, final String default
         }
     }
 
+    CheckStyleResult(final Run<?, ?> build, final BuildHistory history,
+                     final ParserResult result, final String defaultEncoding, final boolean canSerialize) {
+        super(build, history, result, defaultEncoding);
+
+        if (canSerialize) {
+            serializeAnnotations(result.getAnnotations());
+        }

```

Comment:
It was used [here](https://github.com/jenkinsci/checkstyle-plugin/pull/8/files#diff-3d74beb60b0529e460341443b98ab58aR126).

@recena for constructors, it would be better to simply pass through the old constructor to the new one, having:
```
/**
 * @deprecated use {@link #CheckStyleResult(Run, BuildHistory, ParserResult, String, boolean)}
 */
@Deprecated
CheckStyleResult(final AbstractBuild<?, ?> build, final BuildHistory history,
        final ParserResult result, final String defaultEncoding, final boolean canSerialize) {
    this((Run<?, ?>) build, history, result, defaultEncoding, canSerialize);
}

CheckStyleResult(final Run<?, ?> build, final BuildHistory history,
        final ParserResult result, final String defaultEncoding, final boolean canSerialize) {
    super(build, history, result, defaultEncoding);

    if (canSerialize) {
        serializeAnnotations(result.getAnnotations());
    }
}

```